src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -1725,11 +1725,11 @@ private bool ShouldReduceSizeOfCompressionPager(long initialSize, bool forceRedu
             if ((DateTime.UtcNow - _lastCompressionBufferReduceCheck).TotalMinutes < 5)
                 return false;
 
-            if (_maxNumberOfPagesRequiredForCompressionBuffer < _compressionPager.NumberOfAllocatedPages / 2)
+            if (_maxNumberOfPagesRequiredForCompressionBuffer > _compressionPager.NumberOfAllocatedPages / 2)
             {
                 // we recently used at least half of the compression buffer,
                 // no point in reducing size yet, we'll try to do it again next time
-                return true;
+                return false;
```

Comment:
The code here is actually correct. It is stating that we didn't use enough of the file.
The problem is actually different. It is that we are not checking for the maximum size that we should have here.